Snowflake.Data/Core/SFStatement.cs

```diff
@@ -17,6 +17,61 @@
 
 namespace Snowflake.Data.Core
 {
+    /// <summary>
+    /// The status types of the query.
+    /// </summary>
+    public enum QueryStatus
+    {
+        RUNNING,
+        ABORTING,
+        SUCCESS,
+        FAILED_WITH_ERROR,
+        ABORTED,
+        QUEUED,
+        FAILED_WITH_INCIDENT,

```

Comment:
Done

Since the enum values are based on the server, I modified the unit test to use the original values to confirm that they still work with PascalCase

Reply:
You did it by ignoring case. I think that the enum with underscores (e. g. FAILED_WITH_INCIDENT) would not be parsed properly, because ignoring case doesn't solve the issue with uderscores.

Reply:
Currently, the underscores are removed before parsing

The enum is now assigned with a string value instead

Reply:
It would be good to add a unit test for all of the cases. You added the string values as a description.
This introduces a risk that someone would modify it since it may be regarded as a description only.
We could do a unit test for the values.
Also method GetQueryStatusByDescription could be renamed to GetQueryStatusByStringValue since it is a string value rather than a description.
I would also propose to rename class QueryStatuses to QueryStatusExtensions.

Reply:
> It would be good to add a unit test for all of the cases.

There's a unit test for all the statuses [here](https://github.com/snowflakedb/snowflake-connector-net/blob/d4994d2ac8e035bb573daf09e0166e3be4a4dfc0/Snowflake.Data.Tests/UnitTests/SFStatementTest.cs#L141)

Or by case did you mean like lowercase and uppercase?

> You added the string values as a description.
This introduces a risk that someone would modify it since it may be regarded as a description only

It can be changed to something like `StringValue` or `ServerValue` so it's clearer that's the value the server sends back

> We could do a unit test for the values.

Did you mean something like use the original string value and check if it's correctly parsed to the query status?

> Also method GetQueryStatusByDescription could be renamed to GetQueryStatusByStringValue since it is a string value rather than a description.

Done

> I would also propose to rename class QueryStatuses to QueryStatusExtensions.

Done

Reply:
> You used FirstOrDefault() method so in case of unknown string value in the response it would be converted to NO_DATA. Is it for purpose? If so, there should be a test for that. But maybe in case of unknown query status we should rather throw an exception because we actually don't now how to interpret given state?

I used `NO_DATA` as the default because if `RUNNING` was the default value then it would've kept retrying over and over

Adding an extra status to indicate it's unknown would be better since it's clearer if there's actually no data or if the status is just unknown

> Maybe let's make the comparison condition case insensitive.

Done

Reply:
Other than the above, these changes have been committed as well:
- Added Unknown status for queries and throw exception if unknown
- Added test for GetQueryStatusByStringValue in lowercase and uppercase
- Replaced `Description` attribute with `StringAttr` attribute